io/src/vtk_lib_io.cpp

```diff
@@ -74,26 +74,35 @@ pcl::io::loadPolygonFile (const std::string &file_name, pcl::PolygonMesh& mesh)
 }
 
 ///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
-int
-pcl::io::savePolygonFile (const std::string &file_name, const pcl::PolygonMesh& mesh)
+bool
+pcl::io::savePolygonFile (const std::string &file_name,
+                          const pcl::PolygonMesh& mesh,
+                          const bool binary_format)
 {
-  // TODO: what about binary/ASCII modes?!?!?!
   // TODO: what about sensor position and orientation?!?!?!?
-  // TODO: how to adequately catch exceptions thrown by the vtk writers?!
   std::string extension = file_name.substr (file_name.find_last_of (".") + 1);
   if (extension == "pcd") // no Polygon, but only a point cloud
   {
-    int error_code = pcl::io::savePCDFile (file_name, mesh.cloud);
+    int error_code;
+    if (binary_format)
+    {
+      pcl::PointCloud<pcl::PointXYZRGB> cloud;
```

Comment:
What if the mesh cloud contained different channels than XYZRGBA?

But anyway, I'm not sure why you make this cast for the "binary" case. The [savePCDFile()](https://github.com/PointCloudLibrary/pcl/blob/master/io/include/pcl/io/pcd_io.h#L563) function allows to save blob point clouds in binary mode as well.